### PR TITLE
add simons refined samples

### PR DESCRIPTION
**Pull request by Veluchs**
does not include the samples which rely on external test data.

### CODE REVIEW

**Comment by terryyz** [comment]
Hi Simon!

Could you please also add your split to `data/raw` and run the pytest workflows in your forked repo?

To set up the workflows, you need to enable GitHub Actions in your repo and may replace the `script/run.sh` with the following content:

```sh
NAMES=(chien jenny wenhao niklas hanhu ratna simon)
for name in "${NAMES[@]}"; do
    cp data/raw/*"$name"*py data/clean
done
flake8 data/clean/*.py --select=E9,F63,F7,F82 --show-source --statistics
python script/parse.py

for name in "${NAMES[@]}"; do
    for file in data/processed/*"$name"*wo_doc.py; do

        if ! pytest "$file"; then
            echo "Pytest failed on $file, stopping..."
            exit 1
        fi
    done

    for file in data/processed/*"$name"*w_doc.py; do
        if ! pytest --doctest-modules "$file"; then
            echo "Pytest failed on $file, stopping..."
            exit 1
        fi
    done
done
```

Cheers

File: data/clean/f_648_simon.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    - df (pd.DataFrame): The DataFrame to analyze.
+    - height (int): The threshold to filter rows based on the second column in 'columns'.
+    - weight (int): The threshold to filter rows based on the third column in 'columns'.
+    - columns (list of str): A list of column names to use, where the first is the dependent variable.
+
+    Returns:
+    - sm.regression.linear_model.RegressionResultsWrapper: The result of the OLS regression, or None if no rows meet the criteria or DataFrame is empty.
+
+    Requirements:
+    - pandas for DataFrame manipulation.
```
Comment (terryyz) [inline data/clean/f_648_simon.py]: please remove any NL descriptions for the required modules like this one

**Comment by Veluchs** [comment]
Should I add my finalized samples to raw or the initial ones I received?

**Comment by terryyz** [comment]
> Should I add my finalized samples to raw or the initial ones I received?

Finalized ones :)

File: data/clean/f_648_simon.py (hunk 1 above)
Comment (Veluchs) [reply data/clean/f_648_simon.py]: done!

**Comment by Veluchs** [comment]
image

I'm running into this error. Seems to me that there is a file missing. Can you help me out?

**Comment by terryyz** [comment]
> image
> 
> I'm running into this error. Seems to me that there is a file missing. Can you help me out?

Just fixed `script/parse.py`. Please git pull and run the actions again :)

**Comment by Veluchs** [comment]
image

I'm running into this error. The function creates a log file using the logging module. Locally the test run just fine. Any idea why that happens? 
I could probably add a save path for the log file though, I guess that would work.

**Comment by terryyz** [comment]
> image
> 
> I'm running into this error. The function creates a log file using the logging module. Locally the test run just fine. Any idea why that happens? I could probably add a save path for the log file though, I guess that would work.

I don't the specific reasons, unfortunately. However, I'd suggest passing a specific path for the function to write the log file. You can set up the dummy directory in the test cases and tear it down later :)

**Comment by terryyz** [comment]
I notice that there are a few issues when parsing your data:
image
```
data/raw/f_676_simon.py Less than 2 libraries are used

data/raw/f_681_simon.py Less than 2 libraries are used
data/raw/f_726_simon.py Requirements is missing

data/raw/f_720_simon.py Less than 2 libraries are used
data/raw/f_725_simon_chien_edit.py Returns is missing
data/raw/f_647_simon.py Example is missing
data/raw/f_679_simon.py Less than 2 libraries are used
```

Could you please double check and fix them based on the notes? Thanks!

**Comment by Veluchs** [comment]
Hey,

all test conclude successfully now. Should be ready to merge.

**Comment by terryyz** [comment]
Hi @Veluchs, could you please resolve your `data/processed` by updating your repo to the latest version? I also found that you pushed quite a few external data to the repo. Could you also remove them? Thanks!

**Comment by Veluchs** [comment]
> Hi @Veluchs, could you please resolve your `data/processed` by updating your repo to the latest version? I also found that you pushed quite a few external data to the repo. Could you also remove them? Thanks!

Hi @terryyz, 
now it should be ready to merge. Removed the external data which I pushed accidentally and also updated to the latest version.

Only the remaining names need to be added to run.sh

**Pull request merged**